### PR TITLE
column concept annotations

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
New column annotation: principalConceptCode

### CODE REVIEW

File: experiment/resources/schemas/dbscripts/sqlserver/exp-20.009-20.010.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE exp.PropertyDescriptor ADD COLUMN PrincipalConceptCode NVARCHAR(50) NULL;
```
Comment (labkey-adam) [inline experiment/resources/schemas/dbscripts/sqlserver/exp-20.009-20.010.sql]: Remove `COLUMN`; this is illegal on SQL Server and causing failures

File: experiment/resources/schemas/dbscripts/postgresql/exp-20.009-20.010.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE exp.PropertyDescriptor DROP COLUMN PrincipleConceptCode;
```
Comment (labkey-adam) [inline experiment/resources/schemas/dbscripts/postgresql/exp-20.009-20.010.sql]: I don't see where either `PrincipleConceptCode` or `PrincipalConceptCode` gets added on PostgreSQL

File: api/src/org/labkey/api/exp/OntologyManager.java [hunk 3]
```diff
@@ -1703,15 +1705,18 @@ private static int insertPropertyIfNotExists(User user, PropertyDescriptor pd, P
...
+        if (!Objects.equals(pdIn.getLookupSchema(), pd.getLookupSchema()))
             colDiffs.add("LookupSchema");
 
-        if (Objects.equals(pdIn.getLookupQuery(), pd.getLookupQuery()))
+        if (!Objects.equals(pdIn.getLookupQuery(), pd.getLookupQuery()))
             colDiffs.add("LookupQuery");
 
+        if (!Objects.equals(pdIn.getPrincipalConceptCode(), pd.getPrincipalConceptCode()))
+            colDiffs.add("PrincipalConceptCode");
+
```
Comment (labkey-adam) [inline api/src/org/labkey/api/exp/OntologyManager.java:1719]: Wow, looks like this whole diffing code path has been wrong forever

File: api/src/org/labkey/api/data/WrappedColumnInfo.java [hunk 4]
```diff
@@ -901,6 +901,26 @@ public void setIsRootDbSequence(boolean b)
...
+                {
+                    return code;
+                }
+            };
+        }
+
+        @Override
+        public String getPrincipalConceptCode()
+        {
+            return null;
```
Comment (labkey-adam) [inline api/src/org/labkey/api/data/WrappedColumnInfo.java]: Get from delegate?

File: api/src/org/labkey/api/exp/PropertyDescriptor.java [hunk 5]
```diff
@@ -70,6 +70,9 @@
     private boolean _mvEnabled;
     private String _mvIndicatorStorageColumnName;        // only valid if mvEnabled
 
+    // OntologyService related fields
+    private String _principalConceptCode;
```
Comment (labkey-adam) [inline api/src/org/labkey/api/exp/PropertyDescriptor.java]: Unused... remove? I think ColumnRenderPropertiesImpl handles this. Either that or add getter/setter for this.

File: experiment/src/org/labkey/experiment/XarExporter.java [hunk 6]
```diff
@@ -714,6 +714,9 @@ private void addPropertyDescriptor(DomainDescriptorType xDomain, DomainProperty
         if (domainProp.getRangeURI().equals("http://www.w3.org/2001/XMLSchema#string"))
             xProp.setScale(domainProp.getScale());
 
+        if (null != domainProp.getPrincipalConceptCode())
+            xProp.setScale(domainProp.getScale());
```
Comment (labkey-adam) [inline experiment/src/org/labkey/experiment/XarExporter.java]: Looks wrong... should be setting concept code, not scale

**Review by labkey-adam — CHANGES_REQUESTED** [review]

File: api/src/org/labkey/api/data/WrappedColumnInfo.java (hunk 4 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/data/WrappedColumnInfo.java]: Handled in parent class, intellij being 'helpful'

File: experiment/resources/schemas/dbscripts/postgresql/exp-20.009-20.010.sql (hunk 2 above)
Comment (labkey-matthewb) [reply experiment/resources/schemas/dbscripts/postgresql/exp-20.009-20.010.sql]: Dang must have edited a source window instead of a console window!?

**Review by labkey-adam — APPROVED** [review]

**Pull request merged**